dbt/main.py

```diff
@@ -355,7 +355,21 @@ def parse_args(args):
     compile_sub = subs.add_parser('compile', parents=[base_subparser])
     compile_sub.set_defaults(cls=compile_task.CompileTask, which='compile')
 
-    for sub in [run_sub, compile_sub]:
+    docs_sub = subs.add_parser('docs', parents=[base_subparser])
+    docs_subs = docs_sub.add_subparsers()
+    # it might look like docs_sub is the correct parents entry, but that
+    # will cause weird errors about 'conflicting option strings'.
+    generate_sub = docs_subs.add_parser('generate', parents=[base_subparser])
+    generate_sub.set_defaults(cls=generate_task.GenerateTask,
+                              which='generate')
+    generate_sub.add_argument(
+        '--no-compile',
+        action='store_false',
+        dest='compile',
+        help='Do not run "dbt compile" as part of docs generation'
+    )
+
+    for sub in [run_sub, compile_sub, generate_sub]:
```

Comment:
ahhh this is a good idea! this means you can specify models to compile, or exclude models in your dbt docs generate command, right?

Reply:
Yeah. You actually can't even do the `super(...).run()` without this (or adding a bunch of stuff to `self.args` manually, I suppose). But it has some nice consequences like that!